app/liquid_tags/unified_embed/tag.rb

```diff
@@ -51,8 +51,10 @@ def self.validate_link!(link)
       uri = URI.parse(link)
       http = Net::HTTP.new(uri.host, uri.port)
       http.use_ssl = true if http.port == 443
-      path = uri.path.presence || "/"
-      response = http.request_head(path)
+
+      req = Net::HTTP::Get.new(uri.request_uri)
```

Comment:
Hmm, we used to send a HEAD request, we're changing that to a GET request.

 It's possible in some cases HEAD is fast (maybe a db lookup) and GET is slow (render the full content and transfer it).

We're discarding the response (and body) because this is called for side-effect and not for value.

Is it possible to use [Net::HTTP::Head](https://ruby-doc.org/stdlib-3.0.2/libdoc/net/http/rdoc/Net/HTTP/Head.html) here like we had before?

Reply:
side thought - since request_head takes two arguments (the second is headers) the smallest change appears to be setting the user agent in the request_head call (nothing wrong with this version, but we don't _have_ to build a HEAD request to do what we're trying here).

```
[18] pry(main)> response = http.request_head(path, {"User-Agent" => user_agent_string })
=> #<Net::HTTPOK 200 OK readbody=true>                                       
[19] pry(main)> response = http.request_head(path)
=> #<Net::HTTPForbidden 403 Forbidden Bots readbody=true>
[20] pry(main)> uri
=> #<URI::HTTPS https://dev.to/dailydevtips1/8-google-analytics-privacy-focussed-alternatives-3e8h>
```

Reply:
I think the current implementation is more readable 😅  thanks Dan! 👋🏾